strawberryfields/cli/__init__.py

```diff
@@ -0,0 +1,311 @@
+        Please enter the port number to connect with: [443] 12345
+
+        $cat C:\Users\USERNAME\AppData\Local\Xanadu\strawberryfields\config.toml
+        [api]
+        authentication_token = "MYAUTH"
+        hostname = "MYHOST"
+        use_ssl = false
+        port = "12345"
+
+    Returns:
+        dict[str, Union[str, bool, int]]: the configuration options
+    """
+    default_config = create_config()["api"]
+
+    # Getting default values that can be used for as messages when getting inputs
+    hostname_default = default_config["hostname"]
+    ssl_default = "y" if default_config["use_ssl"] else "n"
+    port_default = default_config["port"]
+
+    authentication_token = (
+        input("Please enter the authentication token to use when connecting: [] ")
+    )
+
+    if not authentication_token:
+        sys.stdout.write("No authentication token was provided, please configure again.")
+        sys.exit()
+
+    hostname = (
+        input("Please enter the hostname of the server to connect to: [{}] ".format(hostname_default)) or hostname_default
+    )
+
+    ssl_input = input("Should the client attempt to connect over SSL? [{}] ".format(ssl_default)) or ssl_default
+    use_ssl = ssl_input.upper() == "Y"
+
+    port = input("Please enter the port number to connect with: [{}] ".format(port_default)) or port_default
+
+    kwargs = {
+        "authentication_token": authentication_token,
+        "hostname": hostname,
+        "use_ssl": use_ssl,
+        "port": port,
+    }
+    return kwargs
+
+def run_blackbird_script(args):
+    """Run a blackbird script.
+
+    Related arguments:
+    * input: the input blackbird script to be run
+    * output: the output file to store the results in (optional)
+
+    Args:
+        args (ArgumentParser): arguments that were specified on the command
+            line stored as attributes in an argument parser object
+    """
+    try:
+        program = load(args.input)
+    except FileNotFoundError:
+        sys.stdout.write("The {} blackbird script was not found.".format(args.input))
+        sys.exit()
+
+    eng = StarshipEngine(program.target)
+
+    sys.stdout.write("Executing program on remote hardware...\n")
+    result = eng.run(program)
+
+    if result and result.samples is not None:
+        write_script_results(result.samples, output_file=args.output)
+    else:
+        sys.stdout.write("Ooops! Something went wrong with obtaining the results. Please check the Blackbird script specified and the connection to the remote engine.")
+        sys.exit()
+
+def write_script_results(samples, output_file=None):
+    """Write the results of the script either to a file or to the standard output.
+
+    Args:
+        samples (array[float]): array of samples
+        output_file (str or None): the path to the output file, None if no output was defined
+    """
+    if output_file:
+        with open(output_file, "w") as file:
+            file.write(str(samples))
```

Comment:
Just thinking out loud, but I wonder now if there might be a better or more standard way to serialize the output here. Maybe we could write to an `.npy` file if the specified filename has the extension, or maybe that's trying to be too clever :thinking:

Reply:
I think that's a good idea, we could provide a way for that! Maybe using `np.save` could aid in this. For now, however, I'd rather keep it as is and see if a need for such a change (or potentially an extra option) would be needed.

Reply:
Yes, I agree it's best to leave it as is to avoid the possibility of breaking users' current workflows. :+1: